sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -47,60 +47,113 @@ Ext.define('Slate.cbl.view.student.Dashboard', {
+                var len = demonstrations.length,
+                    x = 0,
+                    i = 0,
+                    demo,
+                    sortedDemonstrations = {},
+                    levelsSorted,
+                    demos,
+                    demosLen,
+                    displayDemonstrations = [],
+                    scored = [],
+                    missed = [];
+                
+                // If no limit is specified, sort all demonstrations
+                limit = isNaN(limit) ? len : limit;
+                
+                // Define functions outside of loops
+                function sortByNewest(a, b) {
+                    // Newest first
+                    return a.Demonstrated > b.Demonstrated ? -1 : a.Demonstrated < b.Demonstrated ? 1 : 0;
+                }
+                
+                function sortByOldest(a, b) {
+                    // Oldest first
+                    return a.Demonstrated > b.Demonstrated ? 1 : a.Demonstrated < b.Demonstrated ? -1 : 0;
+                }
+                
+                // Separate demonstrations by level
+                for (x = 0; len > x; x++) {
+                    demo = demonstrations[x];
+                    sortedDemonstrations[demo.Level] = sortedDemonstrations[demo.Level] || [];
+                    sortedDemonstrations[demo.Level].push(demo);
+                }
+                
+                // Sort the levels of demonstrations observed from greatest to least
+                levelsSorted = Object.keys(sortedDemonstrations).sort(function (a, b) {
+                    a = parseInt(a, 10);
+                    b = parseInt(b, 10);
+                    return a > b ? -1 : a < b ? 1 : 0;
                 });
-
-                blocks = blocks.splice(0, demonstrationsRequired);
-
-                // sort by most recent, followed by M
-                Ext.Array.sort(blocks, function(a, b) {
-                    if (a.Level == 0) {
-                        return 1;
+                
+                // Loop over demonstrations starting with the highest levels
+                levelLoop:
+                for (x = 0, len = levelsSorted.length; len > x; x++) {       
+                    // Sort demonstrations for x-level by date (newest first)
+                    demos = sortedDemonstrations[levelsSorted[x]].sort(sortByNewest);
+            
+                    // Get limit-number of demonstrations
+                    for (i = 0, demosLen = demos.length; demosLen > i; i++) {
+                        displayDemonstrations.push(demos[i]);
+                        if (--limit === 0) {
+                            // Break out early
+                            break levelLoop;
+                        }
                     }
-
-                    if (b.Level == 0) {
-                        return -1;
+                }
+                
+                // Split up scored and missed demonstrations while keeping order by date (oldest first)
```

Comment:
This could be done in the loop above, but if any of the business logic changes, this is much more manageable and easier to understand I think. This last step is critical.